devices/YarpRobotLoggerDevice/tests/YarpRobotLoggerDeviceTest.cpp

```diff
@@ -0,0 +1,233 @@
+// Once this is used in multiple tests, we can move it to a common place to share it among tests
+bool blf_setenv(const std::string &_name, const std::string &_value)
+{
+#ifdef _WIN32
+    if (0 != _putenv_s(_name.c_str(), _value.c_str()))
+    {
+        return false;
+    }
+#else
+    if (0 != ::setenv(_name.c_str(), _value.c_str(), true))
+    {
+        return false;
+    }
+#endif
+    return true;
+}
+
+
+bool ensureYARPAndBLFYARPDevicesCanBeFound()
+{
+    // To make sure that YarpRobotLoggerDevice is found from the build directory, we add CMAKE_BINARY_DIR
+    // to YARP_DATA_DIRS
+#ifdef _WIN32
+    std::string envVarListSeparator = ";";
+#else
+    std::string envVarListSeparator = ":";
+#endif
+
+    // Make sure that YARP devices can be found
+    std::string new_yarp_data_dirs_value = YARP_DATA_INSTALL_DIR_FULL;
+
+    // Make sure that BLF devices are available
+    new_yarp_data_dirs_value = new_yarp_data_dirs_value + envVarListSeparator + CMAKE_BINARY_DIR + "/share/yarp";
+
+    return blf_setenv("YARP_DATA_DIRS", new_yarp_data_dirs_value);
+}
+
+TEST_CASE("Launch simple logger")
+{
+    // This test launched a dummy logger and verify that it effectively saves some data
+    // Locate location of .exe file of ipit-hipac-codesys-embobj-bridge, done in getDirectoryOfExecutable()
+
+    // As a preliminary step, we rename any leftover .md and .mat files contained in the current working directory
+    // to .md.bak and .mat.bak, to ensure that they are not mistakenly files generated by this tests
+    std::filesystem::path currentPath = std::filesystem::current_path();
+    for (const auto& entry : std::filesystem::directory_iterator(currentPath)) {
+        if (entry.is_regular_file()) {
+            std::filesystem::path filePath = entry.path();
+            std::string extension = filePath.extension().string();
+
+            if (extension == ".mat" || extension == ".md") {
+                std::filesystem::path newFilePath = filePath;
+                newFilePath += ".bak";
+                std::filesystem::rename(filePath, newFilePath);
+            }
+        }
+    }
+
+    // We will be opening yarp ports in the tests, but we do not want to run the yarpserver, so let's set YARP
+    // in local mode (everything will work fine as long as all the ports are opened in the same process)
+    yarp::os::Network network;
+    yarp::os::NetworkBase::setLocalMode(true);
+
+    // We ensure that YARP_DATA_DIRS contains the values necessary to find YARP and BLF devices
+    REQUIRE(ensureYARPAndBLFYARPDevicesCanBeFound());
+
+    // We hardcode the yarprobotinterface configuration file is in the source directory and passed via
+    // target_compile_definitions
+    std::filesystem::path pathToXmlConfigurationFile = std::filesystem::path(CMAKE_CURRENT_SOURCE_DIR) / std::filesystem::path("launch-yarp-robot-logger.xml");
+
+    INFO("Loading yarprobotinterface file from " << pathToXmlConfigurationFile.string());
```

Comment:
You may use blf text logging for the prints

Reply:
Done in https://github.com/ami-iit/bipedal-locomotion-framework/pull/862/commits/5c1861f96c0fa6e1ac669e1dd90dbde3f4adafd9, it is a good idea as I never understood when catch actually prints that messages.